src/Log/Migrations/CreateNewLogTable.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Give\Log\Migrations;
+
+use Give\Framework\Database\DB;
+use Give\Framework\Migrations\Contracts\Migration;
+use Give\Framework\Database\Exceptions\DatabaseQueryException;
+use Give\Framework\Migrations\Exceptions\DatabaseMigrationException;
+
+
+
+/**
+ * Class CreateNewLogTables
+ * @package Give\Log\Migrations
+ *
+ * @since 2.9.7
+ */
+class CreateNewLogTable extends Migration {
+
+	/**
+	 * @return string
+	 */
+	public static function id() {
+		return 'create_new_log_table';
+	}
+
+	/**
+	 * @return int
+	 */
+	public static function timestamp() {
+		return strtotime( '2021-01-28 12:00' );
+	}
+
+
+	public function run() {
+		global $wpdb;
+
+		$sql = "CREATE TABLE {$wpdb->prefix}give_log (
+			id BIGINT UNSIGNED NOT NULL AUTO_INCREMENT,
+			log_type VARCHAR(16) NOT NULL,
+			migration_id VARCHAR(64) NULL,
+			data text NOT NULL,
+			category VARCHAR(64) NOT NULL,
+			source VARCHAR(64) NOT NULL,
+			date DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+			PRIMARY KEY  (id),
+			KEY log_type (log_type),
+			KEY migration_id (migration_id),
+			KEY category (category),
+			KEY source (source)
+		) {$wpdb->get_charset_collate()}";
```

Comment:
Go ahead and add the `@method` for get_charset_collate to the `DB` facade. You can just use that.

Reply:
Maybe we could extend this even more and add the `getTable( $name )` method as well? Most of the time we use `global $wpdb` just to get the table name`"{$wpdb->prefix}table_name"` and to me that doesn't look very well.  We can also add a setter method as well, a method that will set the table name as a `$wpdb` property, but I'm not quite sure about that.

Reply:
That's not quite in the scope of the intent of that facade so far. I like where you're going, but I'd want to think more about that and not just squeeze it in this epic.

Reply:
I agree we need to think a bit more about this. 

Resolved: 16724673